transportable-udfs-codegen/src/main/java/com/linkedin/transport/codegen/ProjectContext.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.codegen;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.transport.compile.UDFProperties;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+
+/**
+ * Encapsulates configuration related to a Transport UDF project required by a {@link WrapperGenerator} such as
+ * UDF metadata and the output directories for the wrappers and corresponding resources
+ */
+public class ProjectContext {
+
+  private UDFProperties _udfProperties;
+  private File _sourcesOutputDir;
+  private File _resourcesOutputDir;
+
+  public UDFProperties getUdfProperties() {
+    return _udfProperties;
+  }
+
+  public File getSourcesOutputDir() {
+    return _sourcesOutputDir;
+  }
+
+  public File getResourcesOutputDir() {
+    return _resourcesOutputDir;
+  }
+
+  @VisibleForTesting
+  ProjectContext(UDFProperties udfProperties, File sourcesOutputDir, File resourcesOutputDir) {
+    _udfProperties = udfProperties;
+    _sourcesOutputDir = sourcesOutputDir;
+    _resourcesOutputDir = resourcesOutputDir;
+  }
+
+  public ProjectContext(File udfPropertiesFile, File sourcesOutputDir, File resourcesOutputDir) {
+    try (FileReader reader = new FileReader(udfPropertiesFile)) {
```

Comment:
Extract try/catch to separate method udfPriperties(File) defined in the caller/utils? Push try/catch to forJson itself and make it take File?